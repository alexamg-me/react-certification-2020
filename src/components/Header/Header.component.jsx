import React from 'react';
import styled from 'styled-components';
import Search from '../Search';
import MainMenu from '../MainMenu';
import DarkMode from '../DarkMode';
import UserMenu from '../UserMenu';

export const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  padding: 15px 20px;
  background-color: #70dbcd;
  height: 70px;
  box-shadow: 0 0 10px #409287;
  margin-bottom: 20px;
`;
const Left = styled.div`
  display: flex;
  justify-content: space-between;
`;
const Right = styled.div`
  display: flex;
  justify-content: space-between;
`;

function Header() {
  return (
    <Wrapper>
      <Left>
        <MainMenu />
        <Search />
      </Left>
      <Right>
        <DarkMode />
        <UserMenu />
      </Right>
    </Wrapper>
  );
}

export default Header;
