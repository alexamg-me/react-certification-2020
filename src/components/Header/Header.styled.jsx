import styled from 'styled-components';

const HeaderWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  padding: 15px 20px;
  background-color: #70dbcd;
  height: 70px;
  box-shadow: 0 0 10px #409287;
  margin-bottom: 20px;
`;
const Left = styled.div`
  display: flex;
  justify-content: space-between;
`;
const Right = styled.div`
  display: flex;
  justify-content: space-between;
`;

export { HeaderWrapper, Left, Right };
