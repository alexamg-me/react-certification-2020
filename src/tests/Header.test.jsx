import React from 'react';
import TestRenderer from 'react-test-renderer';
import Header from '../components/Header/Header.component';

describe(`Component: Header`, () => {
  test(`Header renders with default props`, () => {
    const component = TestRenderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
