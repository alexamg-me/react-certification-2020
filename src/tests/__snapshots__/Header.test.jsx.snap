// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header Header renders with default props 1`] = `
<div
  className="sc-crrsfI qrNWt"
>
  <div
    className="sc-dQppl ea-dLjL"
  >
    <div
      className="sc-hKgILt hpfQOr"
    >
      <div
        className="sc-eCssSg fSuOCU"
      >
        <div
          className="sc-jSgupP eKSphI"
        />
        <div
          className="sc-jSgupP eKSphI"
        />
        <div
          className="sc-jSgupP eKSphI"
        />
      </div>
    </div>
    <div
      className="sc-bdfBwQ ffLsOo"
    >
      <div
        className="sc-gsTCUz HFjtR"
      />
      <input
        className="sc-dlfnbm ifQijI"
        type="text"
      />
    </div>
  </div>
  <div
    className="sc-bqyKva duXVGw"
  >
    <div
      className="sc-gKsewC gcycVY"
    >
      <div
        className="sc-iBPRYJ gqNZGR"
      >
        <div
          className="sc-fubCfw gWoYKE"
        />
        <div
          className="sc-pFZIQ xJzqo"
        />
      </div>
      <div
        className="sc-jrAGrp bqZhEl"
      >
        Dark Mode
      </div>
    </div>
    <div
      className="sc-kEjbxe cpepa-d"
    >
      <div
        className="sc-iqHYGH fIAESH"
      />
    </div>
  </div>
</div>
`;
